### ISSUE TITLE
ani-cli not working on macOS Monterey

### ISSUE THREAD

**Issue by shadows1003929** [pystardust/ani-cli#205]
ani-cli not opening mpv even when mpv is installed

macOS Monterey
MacBook Air 2017

**Comment by insperace** [comment]
I had to :
`brew install util-linux
export PATH="$PATH:/usr/local/opt/util-linux/bin"`

**Comment by insperace** [comment]
> ani-cli not opening mpv even when mpv is installed
> 
> macOS Monterey MacBook Air 2017

I had to :
`brew install util-linux `
`export PATH="$PATH:/usr/local/opt/util-linux/bin"`

**Comment by shadows1003929** [comment]
ok

**Comment by leaf-ts** [comment]
I also have an issue with ani-cli not working on Monterey.
For me mpv does not launch at all, was the same for you?
mpv seems to be able to launch properly and play any other local file
the only thing I can see at the log is 
`[   0.026][v][file] Opening 
[   0.026][e][file] Cannot open file '': No such file or directory
[   0.026][e][stream] Failed to open .`

**Comment by ura43** [comment]
Is this a duplicate of #123 ?

**Comment by leaf-ts** [comment]
> 

no, at least it does not seem like it, or any of the input in that discussion fixed it.
I think it is relevant with other threads that the video does not play because of a change in gogoanime or something similar

**Comment by Prinsodular** [comment]
> ani-cli not opening mpv even when mpv is installed
> 
> macOS Monterey MacBook Air 2017

This might be linked with PR #137. Try replacing line 280 with:
`nohup $player_fn $play_link > /dev/null 2>&1
                        echo "${c_green}\nVideo played"`

**Comment by ura43** [comment]
@leaf-ts can you test if PR #137 resolves your issue?
If it does, this would be a duplicate of #2

**Comment by ura43** [comment]
> > 
> 
> no, at least it does not seem like it, or any of the input in that discussion fixed it. I think it is relevant with other threads that the video does not play because of a change in gogoanime or something similar

Well, those bugs were fixed since you last commented.
Disregard my last comment, PR #137 is built on an old master that doesn't have those bugs fixed yet.
Check if the workarounds in #2 work for you while its #137 isn't merged yet

**Comment by leaf-ts** [comment]
@ura43  unfortunately still nothing. 
I have attached a debug for you if that help
[debug.log](https://github.com/pystardust/ani-cli/files/7792585/debug.log)
s

**Comment by ura43** [comment]
Sad, I will ping you once #137 is merged. 
It doesn't work for you currently because you have to choose between 2 huge bugs or 1 mac bug and either choice results in stuff not working

**Comment by leaf-ts** [comment]
@ura43 thank you very much.
if you want me to do any tests etc please let me know :)

**Comment by ura43** [comment]
@leaf-ts can you test if it works now?

**Comment by leaf-ts** [comment]
> @leaf-ts can you test if it works now?
question, when I try to "sudo make" I get this error
sed -i '2s}.*}ORIGINAL_DIR="/Users/....../ani-cli"}' ./ani-cli
sed: 1: "./ani-cli": invalid command code .
make: *** [install] Error 1

any ideas?

**Comment by ura43** [comment]
Oof, sorry we broke things for mac users again.
See [this](https://stackoverflow.com/questions/21470124/invalid-command-code-error-from-sed-after-running-find-and-sed-on-mavericks) stackoverflow post.
Replace the sed line with the following:  `sed -i "" '2s}.*}ORIGINAL_DIR="/Users/....../ani-cli"}' ./ani-cli`

**Comment by leaf-ts** [comment]
@ura43 
so, the sed line works but then the "ln -sr" does not.
if I remove the "r" the make completes fine, although the link is no longer relative I guess?
the error is 
```
ln -sr ani-cli /usr/local/bin/ani-cli
ln: illegal option -- r
usage: ln [-Ffhinsv] source_file [link_name]
       ln [-Ffhinsv] source_file ... linkname_dir
       link source_file link_name
```
something for your next update I guess  ;)

either way if I run "./ani-cli" the app runs fine and the video also plays fine, so that is good!!

**Comment by ura43** [comment]
That is worse news than it might initially seem:
using ln without -r on linux makes the link broken.
But ln -r is an illegal option on mac.
This essentially means that we would have to either get rid of the makefile entirely or set up a seperate mac branch

**Comment by leaf-ts** [comment]
how about if we add a check for the OS and it will run the appropriate common depending on the result?
although not sure if this will complicate things as well as the the others issues you refer to on #255

**Comment by Dink4n** [comment]
How about a simpler solution:
Don't symlink just copy the file

**Comment by ura43** [comment]
> How about a simpler solution: Don't symlink just copy the file

Seems like thats where we'll go
